source/portable/NetworkInterface/STM32Hxx/NetworkInterface.c

```diff
@@ -203,6 +203,34 @@ const PhyProperties_t xPHYProperties =
 };
 /*-----------------------------------------------------------*/
 
+static void prvMACAddressConfig( ETH_HandleTypeDef * heth,
+                                 uint32_t ulIndex,
+                                 const uint8_t * Addr )
+{
+    uint32_t ulTempReg;
+    uint32_t ulETH_MAC_ADDR_HBASE = ( uint32_t ) &( heth->Instance->MACA0HR );
+    uint32_t ulETH_MAC_ADDR_LBASE = ( uint32_t ) &( heth->Instance->MACA0LR );
+
+    /* ETH_MAC_ADDRESS0 reserved for the primary MAC-address. */
+    configASSERT( ulIndex >= ETH_MAC_ADDRESS1 );
+
+    /* ETH_MAC_ADDRESS0 - ETH_MAC_ADDRESS3 */
+    configASSERT( ulIndex <= ETH_MAC_ADDRESS3 );
```

Comment:
Please add a comment also that more than 4 MAC addresses should not enabled simultaneously.